src/views/author_notes.ts

```diff
@@ -14,14 +14,114 @@
  * limitations under the License.
  */
 
-import { BaseNodeProps, BaseNodeView } from './base_node_view'
+import { ManuscriptNode } from '@manuscripts/transform'
+
+import BlockView from './block_view'
 import { createNodeView } from './creators'
+import { EditableBlockProps } from './editable_block'
 
-export class AuthorNotesView extends BaseNodeView<BaseNodeProps> {
+export class AuthorNotesView extends BlockView<EditableBlockProps> {
   container: HTMLElement
+  allowedTypes = ['corresp', 'footnote', 'paragraph']
+  headersStrings: { [key: string]: string } = {
+    corresp: 'Correspondence',
+    footnote: 'Author notes',
+    paragraph: 'Author notes',
+  }
 
   public ignoreMutation = () => true
   public stopEvent = () => true
+
+  public initialise = () => {
+    this.createDOM()
+    this.createGutter('block-gutter', this.gutterButtons().filter(Boolean))
+    this.createElement()
+    this.createGutter(
+      'action-gutter',
+      this.actionGutterButtons().filter(Boolean)
+    )
+    this.updateContents()
+  }
+
+  public createElement = () => {
+    this.container = document.createElement('div')
+    this.container.classList.add('block', 'author-notes-container')
+    this.dom.appendChild(this.container)
+  }
+
+  public createDOM = () => {
+    this.dom = document.createElement('div')
+    this.dom.classList.add('block-container')
+    this.dom.classList.add(`block-${this.node.type.name}`)
+    this.dom.setAttribute('contenteditable', 'false')
+  }
+
+  public updateContents = () => {
+    this.container.innerHTML = ''
+
+    const items: { [key: string]: ManuscriptNode[] } = {}
+
+    this.view.state.doc.descendants((node, pos) => {
+      if (node.type.name === 'author_notes') {
+        node.descendants((anNode, anPos) => {
+          items[anNode.type.name] = []
+          if (this.allowedTypes.includes(anNode.type.name)) {
+            items[anNode.type.name].push(anNode)
+            return false
+          }
+        })
+        return false
```

Comment:
In this logic you keep Initializing the array for a node type before checking the condition in the second line might result in an empty array. and I think you will end up with one item for each array.
Also you are preserving memory space for unwanted nodeTypes.

I think you can have a helper such as the below one, and you can use it to determine showing the header for the allowedTypes or not, and you can use it also to build the items array:
export function hasNodeType(node: Node, typeName: string): boolean {
  let found = false;

  node.descendants((childNode) => {
    if (childNode.type.name === typeName) {
      found = true;
      return false; // Stop traversal
    }
    return true; // Continue traversal
  });

  return found;
}

Reply:
There is not title for every (allowed) type. 
Yes, the declaration items[anNode.type.name] = [] is wrong here. For now it is declared only for allowed types: this.allowedTypes.forEach((type) => (items[type] = []))  ...line47